src/audio/eq_fir/fir_hifi3.c

```diff
@@ -32,30 +32,35 @@ void fir_reset(struct fir_state_32x16 *fir)
 	 */
 }
 
-size_t fir_init_coef(struct fir_state_32x16 *fir,
-		     struct sof_eq_fir_coef_data *config)
+int fir_delay_size(struct sof_eq_fir_coef_data *config)
+{
+	/* Check FIR tap count for implementation specific constraints */
+	if (config->length > SOF_EQ_FIR_MAX_LENGTH || config->length < 4)
+		return -EINVAL;
+
+	/* The optimization requires the tap count to be multiple of four */
+	if (config->length & 3)
```

Comment:
Always mask with base 16 i.e. 0x3

Reply:
Yep agree, it improves code readability in if statement to see what's meant.

Reply:
Done (hifi3 & hifi2ep)!